src/main/resources/embedding-providers-config.yaml

```diff
@@ -74,6 +75,28 @@ stargate:
               help: "The API version to use for this operation. This follows the YYYY-MM-DD format."
           properties:
             max-input-length: 16
+          models:
+            - name: text-embedding-3-small
+              parameters:
+                - name: vectorDimension
+                  type: number
+                  required: true
+                  default-value: 512
```

Comment:
Are we hard-coding defaults ourselves? [This page on the OpenAI docs](https://platform.openai.com/docs/guides/embeddings/what-are-embeddings) indicates that text-embedding-3-small's default dimension is 1536

![image](https://github.com/stargate/data-api/assets/1620265/e073b5e5-bbf6-4b31-9b7b-adba34fb09fa)

Reply:
I copied this from non-Azure OpenAI so not sure about reasoning.

@maheshrajamani @Hazel-Datastax WDYT?

Reply:
I _think_ default in these cases would be what is given as the baseline in UI (Portal); neither Data API nor Embedding gateway would implement defaulting logic, but always pass explicit value used with `createCollection()` call.

Reply:
I copied the config from Aaron's design docs and didn't check the openai's docs :(
Yes, Tatu is correct. No matter what the default value is, users have to give one when they create the collection. But we can still change the default value to align with the openai's docs now, it will not cause any trouble.

Reply:
I asked on team Slack on which values to use: we do not have to use same value as OpenAI, if we think users would be served better by lower value. But this is sort of more Product question and I am not sure of the right answer myself. Will see if anyone has strong opinions.

Reply:
Leaving as-is for (near-)future improvements.